pandas/plotting/_matplotlib/core.py

```diff
@@ -1177,7 +1177,11 @@ def _plot_colorbar(self, ax: Axes, **kwds):
         # use the last one which contains the latest information
         # about the ax
         img = ax.collections[-1]
-        return self.fig.colorbar(img, ax=ax, **kwds)
+        with warnings.catch_warnings():
+            # https://github.com/matplotlib/matplotlib/issues/23614
+            # False positive deprecation warning until matplotlib=3.6
+            warnings.filterwarnings("ignore", "Auto-removal of grids")
```

Comment:
is this unnecessary now that our min version is 3.6.1?

Reply:
Yeah I think based on https://github.com/matplotlib/matplotlib/issues/23614#issuecomment-1231935245 should be safe to remove